src/sage/rings/function_field/drinfeld_modules/homset.py

```diff
@@ -421,3 +425,264 @@ def _element_constructor_(self, *args, **kwds):
         return DrinfeldModuleMorphism(self, *args, **kwds)
+
+    def element(self, degree):
+        r"""
+        Return an element of the space of morphisms between the domain and
+        codomain. By default, chooses an element of largest degree less than
+        or equal to the parameter `degree`.
+
+        INPUT:
+
+        - ``degree`` -- the maximum degree of the morphism
+
+        OUTPUT: a univariate ore polynomials with coefficients in `K`
+
+        EXAMPLES::
+
+            sage: Fq = GF(2)
+            sage: A.<T> = Fq[]
+            sage: K.<z> = Fq.extension(3)
+            sage: psi = DrinfeldModule(A, [z, z + 1, z^2 + z + 1])
+            sage: phi = DrinfeldModule(A, [z, z^2 + z + 1, z^2 + z])
+            sage: H = Hom(phi, psi)
+            sage: M = H.element(3)
+            sage: M_poly = M.ore_polynomial()
+            sage: M_poly*phi.gen() - psi.gen()*M_poly
+            0
+
+        ALGORITHM:
+
+            We scan the basis for the first element of maximal degree
+            and return it.
+        """
+        basis = self.Fq_basis(degree)
+        elem = basis[0]
+        max_deg = elem.ore_polynomial().degree()
+        for basis_elem in basis:
+            if basis_elem.ore_polynomial().degree() > max_deg:
+                elem = basis_elem
+                max_deg = elem.ore_polynomial().degree()
+        return elem
+
+    def Fq_basis(self, degree):
+        r"""
+        Return a basis for the `\mathbb{F}_q`-space of morphisms from `phi` to
+        a Drinfeld module `\psi` of degree at most `degree`. A morphism
+        `\iota: \phi \to psi` is an element `\iota \in K\{\tau\}` such that
+        `iota \phi_T = \psi_T \iota`. The degree of a morphism is the
+        `\tau`-degree of `\iota`.
+
+        INPUT:
+
+        - ``degree`` -- the maximum degree of the morphisms in the span.
+
+        OUTPUT: a list of Drinfeld module morphisms.
+
+        EXAMPLES::
+
+            sage: Fq = GF(2)
+            sage: A.<T> = Fq[]
+            sage: K.<z> = Fq.extension(3)
+            sage: psi = DrinfeldModule(A, [z, z + 1, z^2 + z + 1])
+            sage: phi = DrinfeldModule(A, [z, z^2 + z + 1, z^2 + z])
+            sage: H = Hom(phi, psi)
+            sage: B = H.Fq_basis(3)
+            sage: M = B[0]
+            sage: M_poly = M.ore_polynomial()
+            sage: M_poly*phi.gen() - psi.gen()*M_poly
+            0
+
+        ALGORITHM:
+
+            We return the basis of the kernel of a matrix derived from the
+            constraint that `\iota \phi_T = \psi_T \iota`. See [Wes2022]_ for
+            details on this algorithm.
+        """
+        domain, codomain = self.domain(), self.codomain()
+        Fq = domain._Fq
+        K = domain.base_over_constants_field()
+        q = Fq.cardinality()
+        char = Fq.characteristic()
+        r = domain.rank()
+        n = K.degree(Fq)
+        # shorten name for readability
+        d = degree
+        qorder = logb(q, char)
```

Comment:
Here, you don't need to import the function `logb`, you can simply do:
```suggestion
        qorder = char.log(q)
```

Reply:
[TIL](https://dictionary.cambridge.org/dictionary/english/til)